fml/unique_fd.h

```diff
@@ -22,10 +25,48 @@ namespace internal {
 
 namespace os_win {
 
+struct DirCacheEntry {
+  std::wstring filename;
+  FILE_ID_128 id;
+};
+
+// The order of these is important.  Must come before UniqueFDTraits struct
+// else linker error.  Embedding in struct also causes linker error.
+
 struct UniqueFDTraits {
+  inline static std::mutex file_map_mutex;
+  inline static std::map<HANDLE, DirCacheEntry> file_map;
```

Comment:
This must be OOL and be present in a single translation unit.

Reply:
In fact, all the implementations here need to be OOL. Perhaps this is why you were running into linker issues?

Reply:
What I found was, with OOL it doesn't link / function correctly:

1. the map and mutex would not link correctly as member of the struct, only outside
2. the map and mutex would not link correctly if placed after the struct definition, only if placed before
3. At runtime, the delete method would not see the correct memory for the map: store and retrieve methods would both work correctly; however the delete would show a map with 0 items despite there being items there (witness by testing retrieve after delete) which suggested to me that the linker had referenced an incorrect address for some reason.

All 3 of the above issues were fixed by inlining both variables.  Open to suggestions for an alternative fix.

Reply:
Please reopen if there is an alternative viable solution that you'd be more happy with.

Reply:
Pushed a patch with a fix.